### ISSUE TITLE
RxJava 1.1.4 regression crash

### ISSUE THREAD

**Issue by FranciscoE-Hudl** [ReactiveX/RxJava#3911]
After updating to RxJava 1.1.4 some of my chains have started throwing MissingBackpressureException for some operations.

Code:

````
Observable.zip(
            mFirstObs.first(), mSecondObs.first(), mThirdObs.first(), mFourthObs.first(),
            zipLatestState())
    .doOnNext(logPreAggregation())
    .map(applyAggregation());


Func4<A, B, C, D> zipLatestState() {
    return (one, two, three, four) -> Quartet.with(one, two, three, four);
}
````

Error:

```
rx.exceptions.MissingBackpressureException caused by:

OnError while emitting onNext value: org.javatuples.Quartet.class
```

Stacktrace:

```
05-05 11:15:31.423 19767-19767/com.my.app W/System.err: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: org.javatuples.Quartet.class
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.exceptions.Exceptions.throwOrReport(Exceptions.java:192)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:82)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorZip$Zip.tick(OperatorZip.java:264)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorZip$Zip$InnerSubscriber.onNext(OperatorZip.java:335)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.producers.SingleProducer.request(SingleProducer.java:65)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.Subscriber.setProducer(Subscriber.java:211)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:111)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorTake$1.onNext(OperatorTake.java:80)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.operators.NotificationLite.accept(NotificationLite.java:150)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at com.jakewharton.rxrelay.RelaySubscriptionManager$RelayObserver.accept(RelaySubscriptionManager.java:311)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at com.jakewharton.rxrelay.RelaySubscriptionManager$RelayObserver.emitLoop(RelaySubscriptionManager.java:282)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at com.jakewharton.rxrelay.RelaySubscriptionManager$RelayObserver.emitFirst(RelaySubscriptionManager.java:259)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at com.jakewharton.rxrelay.BehaviorRelay$1.call(BehaviorRelay.java:81)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at com.jakewharton.rxrelay.BehaviorRelay$1.call(BehaviorRelay.java:79)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at com.jakewharton.rxrelay.RelaySubscriptionManager.add(RelaySubscriptionManager.java:100)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at com.jakewharton.rxrelay.RelaySubscriptionManager.call(RelaySubscriptionManager.java:59)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at com.jakewharton.rxrelay.RelaySubscriptionManager.call(RelaySubscriptionManager.java:36)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
05-05 11:15:31.423 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.Observable.unsafeSubscribe(Observable.java:8452)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorZip$Zip.start(OperatorZip.java:214)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorZip$ZipSubscriber.onNext(OperatorZip.java:156)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorZip$ZipSubscriber.onNext(OperatorZip.java:122)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.util.ScalarSynchronousObservable$WeakSingleProducer.request(ScalarSynchronousObservable.java:268)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.Subscriber.setProducer(Subscriber.java:209)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.util.ScalarSynchronousObservable$1.call(ScalarSynchronousObservable.java:79)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.util.ScalarSynchronousObservable$1.call(ScalarSynchronousObservable.java:75)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.Observable.unsafeSubscribe(Observable.java:8452)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:248)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:145)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorMap$MapSubscriber.onNext(OperatorMap.java:74)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OnSubscribeRefCount$2.onNext(OnSubscribeRefCount.java:124)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorPublish$PublishSubscriber.dispatch(OperatorPublish.java:587)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at rx.internal.operators.OperatorPublish$PublishSubscriber.onNext(OperatorPublish.java:285)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at com.jakewharton.rxbinding.view.ViewClickOnSubscribe$1.onClick(ViewClickOnSubscribe.java:23)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at android.view.View.performClick(View.java:5242)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at android.widget.TextView.performClick(TextView.java:10540)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at android.view.View$PerformClick.run(View.java:21179)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at android.os.Handler.handleCallback(Handler.java:739)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:95)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at android.os.Looper.loop(Looper.java:145)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:6873)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at java.lang.reflect.Method.invoke(Method.java:372)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
05-05 11:15:31.428 19767-19767/com.my.app W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
```

**Comment by akarnokd** [comment]
The log tells that the `logPreAggregation()`'s return value crashed but the cause is missing from your post. There is a backpressure bug in the operator `filter` that can cause `MissingBackpressureException` but not sure this is related to that particular bug.

**Comment by artem-zinnatullin** [comment]
@FranciscoE-Hudl please report if next patch release resolves your issue, see #3910. Release should be [available soon](https://github.com/ReactiveX/RxJava/pull/3913).

**Comment by FranciscoE-Hudl** [comment]
@akarnokd But it doesn't crash on 1.1.3 and the logs have been working fine (?)

**Comment by FranciscoE-Hudl** [comment]
@akarnokd @artem-zinnatullin Seems to be fixed in 1.1.5
